applications/ecallmgr/src/ecallmgr_call_control.erl

```diff
@@ -192,19 +177,82 @@ fs_nodedown(Srv, Node) ->
-                 ,initial_ccvs=CCVs
-                 }}.
+-spec init(list()) -> {'ok', state()}.
+init(_) ->
+    lager:error("HEY!!"),
+    {'ok', #state{}}.
+
+-spec init_control(pid() , map()) -> 'ok'.
+init_control(Pid, #{node := Node
+                   ,call_id := CallId
+                   ,callback := Fun
+                   ,fetch_id := FetchId
+                   }=Payload) ->
+    proc_lib:init_ack(Pid, {'ok', self()}),
+    Name = ?CALL_CTL_NAME(CallId),
+    try Fun(Payload) of
+        {'ok', #{controller_q := ControllerQ
+                ,controller_p := ControllerP
+                ,control_q := ControlQ
+                ,initial_ccvs := CCVs
+                }} ->
+            bind(Node, CallId),
+            TRef = erlang:send_after(?SANITY_CHECK_PERIOD, self(), 'sanity_check'),
+            State = #state{node=Node
+                          ,call_id=CallId
+                          ,command_q=queue:new()
+                          ,start_time=os:timestamp()
+                          ,sanity_check_tref=TRef
+                          ,fetch_id=FetchId
+                          ,controller_q=ControllerQ
+                          ,controller_p=ControllerP
+                          ,control_q=ControlQ
+                          ,initial_ccvs=CCVs
+                          ,is_node_up=true
+                          ,is_call_up=true
+                          },
+            call_control_ready(State),
+            register(Name, self()),
+            gen_server:enter_loop(?MODULE, [], State, {'local', Name});
+        _Other ->
+            lager:debug("INIT_CONTROL5")
+    catch
+        _Ex:_Err ->
+            lager:debug("BINDINGS ~p : ~p", [_Ex, _Err]),
+            kz_util:log_stacktrace()
+    end,
+    lager:debug("INIT_CONTROL_EXIT"),
+    'ok';
+init_control(Pid, #{node := Node
+                   ,call_id := CallId
+                   ,fetch_id := FetchId
+                   ,controller_q := ControllerQ
+                   ,controller_p := ControllerP
+                   ,control_q := ControlQ
+                   ,initial_ccvs := CCVs
+                   }) ->
+    proc_lib:init_ack(Pid, {'ok', self()}),
+    Name = ?CALL_CTL_NAME(CallId),
+    bind(Node, CallId),
+    TRef = erlang:send_after(?SANITY_CHECK_PERIOD, self(), 'sanity_check'),
+    State = #state{node=Node
+                  ,call_id=CallId
+                  ,command_q=queue:new()
+                  ,start_time=os:timestamp()
+                  ,sanity_check_tref=TRef
+                  ,fetch_id=FetchId
+                  ,controller_q=ControllerQ
+                  ,controller_p=ControllerP
+                  ,control_q=ControlQ
+                  ,initial_ccvs=CCVs
+                  ,is_node_up=true
+                  ,is_call_up=true
+                  },
+    call_control_ready(State),
+    register(Name, self()),
+    gen_server:enter_loop(?MODULE, [], State, {'local', Name}),
+    lager:debug("INIT_CONTROL_EXIT"),
```

Comment:
same.
Also, AFAIK `gen_server:enter_loop` doesn't return so I don't know that this code will be executed.